src/ui/pages/zcl_abapgit_gui_page_sett_remo.clas.abap

```diff
@@ -200,27 +211,38 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_SETT_REMO IMPLEMENTATION.
 
   METHOD choose_branch.
 
-    DATA:
-      lv_url         TYPE zif_abapgit_persistence=>ty_repo-url,
-      lv_branch_name TYPE zif_abapgit_persistence=>ty_repo-branch_name,
-      ls_branch      TYPE zif_abapgit_git_definitions=>ty_git_branch.
+    DATA lv_url         TYPE zif_abapgit_persistence=>ty_repo-url.
+    DATA lv_branch_name TYPE zif_abapgit_persistence=>ty_repo-branch_name.
+    DATA ls_branch      TYPE zif_abapgit_git_definitions=>ty_git_branch.
 
-    IF mo_form_data->get( c_id-offline ) = abap_true.
-      RETURN.
-    ENDIF.
+    IF iv_is_return = abap_false.
 
-    lv_url = mo_form_data->get( c_id-url ).
-    lv_branch_name = zif_abapgit_definitions=>c_git_branch-heads_prefix && mo_form_data->get( c_id-branch ).
+      IF mo_form_data->get( c_id-offline ) = abap_true.
```

Comment:
this is a 

```abap
IF sdf.
  ...
ELSE.
  IF ffds.
     ...
  ENDIF.
ENDIF.
```

it can be changed to

```abap
IF sdf.
 ...
ELSEIF ffds.
  ...
ENDIF.
```

Reply:
just posted a separate comment on that above. Could you pls have a look? But will fix if you insist, no prob

Reply:
is your agument that
```abap
ELSE.
  IF something.
  ENDIF.
ENDIF.
```
is easier to understand/less effort than following?
```abap
ELSEIF something.
ENDIF.
```

Reply:
essentially yes. If the upper `if` has a clear binary nature. IMHO :)

Reply:
For me it depends. 

```abap
IF color = red. 
ELSEIF color = blue.
ELSE.
  IF shape = circle.
  ELSEIF shape = square. 
  ELSE.
  ENDIF. 
ENDIF. 
```

This is easier to follow than the alternative (one if) because I can better understand what "else" means (ie other colors) before checking shapes. 

In general the rule makes sense but there are exceptions where separating conditions into statements can make it better. 🤷‍♀️

Reply:
I guess that code would be better with a CASE

anyhow, for binary

IF binary.
```abap
IF binary.
ELSE.
  IF something.
  ENDIF.
ENDIF.
```

perhaps do

```abap
CASE binary.
WHEN true.
WHEN false.
  IF something.
  ENDIF.
ENDCASE.
```

?

Reply:
yea, well, that also fixes it, at least for now(hope to get some more rules done sometime)

Reply:
I decided not to block the PR so fixed it this way. The question with rules is more a discussion. But i'd keep it somewhere:) maybe an issue?

Maybe my "binary" comment was too abstract. Marc's comment actually shows a similar situation better. **The readability decreases if contexts are mixed**. In Marc's example it is color and shape. 2 independent contexts. In my case it is "initation- return" and "examining the return results". So for discussion-sake I still believe this linter rule makes harm.

Reply:
for Marcs case I think using CASE would fit, triggering https://rules.abaplint.org/change_if_to_case/

just `IF binary ELSE` doesnt trigger that rule

but CASE clearly communicates 2 different scenarios to the reader IMHO